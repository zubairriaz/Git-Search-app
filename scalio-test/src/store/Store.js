import React,{useContext} from 'react';
import request from '../utils/request';

const Store = React.createContext();

export const useStore = () => useContext(Store);

export const StoreProvider = ({ children, initialState, reducer }) => {
  const [globalState, dispatch] = React.useReducer(reducer, initialState);
  const apiRequest = request(dispatch)

  return (
    <Store.Provider value={[globalState, dispatch, apiRequest]}>{children}</Store.Provider>
  );
};