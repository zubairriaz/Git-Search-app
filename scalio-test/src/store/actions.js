import {SET_ITEM, SET_ISLOADING, SET_ERROR_MESSAGE} from '../utils/constants'

export const setItems = (item) => ({
    type: SET_ITEM,
    item,
  });

  export const setErrorMessgae = (message) => ({
    type: SET_ERROR_MESSAGE,
    message,
  });

  export const setIsLoading = (bool) => ({
    type: SET_ISLOADING,
    bool,
  });
  
  