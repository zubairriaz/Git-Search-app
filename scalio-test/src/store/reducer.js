import { SET_ITEM, SET_ISLOADING, SET_ERROR_MESSAGE } from "../utils/constants";
export const initialState = {
	item: {},
	isLoading: false,
	errorMessage: "",
};


export const appReducer = (state = initialState, action) => {
	if (action.type === SET_ITEM) {
		return {
			...state,
			item: { ...action.item },
			isLoading: false,
      errorMessage:""
		};
	}
	if (action.type === SET_ERROR_MESSAGE) {
		return {
			...state,
			errorMessage: extractMessage(action.message),
      item:{items:[],total_count:0},
			isLoading: false,
		};
	}
	if (action.type === SET_ISLOADING) {
		return {
			...state,
			isLoading: action.bool,
		};
	}
};


const extractMessage=res=>res? res.data? res.data.message : "" :"";