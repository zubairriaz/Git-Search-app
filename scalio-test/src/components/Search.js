/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import React from "react";
import { url } from "../utils/constants";
import { useStore } from "../store/Store";

export default function Search({ onChangeText, searchTerm, errorMessage, onSubmitForm }) {
return (
		<React.Fragment>
			<form onSubmit={onSubmitForm}>
				<input
					placeholder={"Enter Login Text"}
					css={getInputCss(errorMessage)}
					type="text"
					name="name"
					value={searchTerm}
					onChange={(event) => onChangeText(event)}
				/>

				<input
				    data-testid="form"
					css={getButtonCss()}
					type="submit"
					value="Submit"
				/>
			</form>
			{errorMessage && (
				<span
					css={css`
						font-size: 12px;
						color: red;
					`}
				>
					{errorMessage}
				</span>
			)}
		</React.Fragment>
	);
}


const getInputCss = (error)=>{
	let color  =  error ? 'red':'#ccc'
	return css`	width: 75%;
	margin: 8px 0;
	display: inline-block;
	border: 1px solid ${color};
	box-shadow: inset 0 1px 3px #ddd;
	border-radius: 4px;
	-webkit-box-sizing: border-box;
	-moz-box-sizing: border-box;
	box-sizing: border-box;
	padding-left: 20px;
	padding-right: 20px;
	padding-top: 12px;
	padding-bottom: 12px;`
}

const getButtonCss = ()=>{
	return css`
	background-color: royalblue;
	border: none;
	color: white;
	padding: 13px 32px;
	text-decoration: none;
	margin: 4px 2px;
	cursor: pointer;
	`
}