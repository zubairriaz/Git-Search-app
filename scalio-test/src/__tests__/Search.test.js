import React from 'react';

import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event'

import { shallow } from 'enzyme';

import Search from '../components/Search';
import {StoreProvider} from "../store/Store"

test('Check If Search Loads Successfully', () => {
  const search = shallow(<StoreProvider><Search/></StoreProvider>);
  expect(search).toMatchSnapshot();
});

test('Typing In Input Renders Correct Result', async () => {
  render(<StoreProvider><Search onChangeText ={()=>{}}/></StoreProvider>);
  userEvent.type(screen.queryByPlaceholderText(/Enter Login Text/i),"hello")
  await waitFor(() => {
     expect(screen.queryByPlaceholderText(/Enter Login Text/i)).toHaveValue("hello");
  });
});




