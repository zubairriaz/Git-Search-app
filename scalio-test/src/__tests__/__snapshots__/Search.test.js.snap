// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check If Search Loads Successfully 1`] = `ShallowWrapper {}`;
